### PR TITLE
Add MatMul operator to patchedast

### LINKED ISSUE #496: Rename fails with missing KeyError: MatMul in patchedast
Ticket originally reported by @eivindjahren in ropevim (https://github.com/python-rope/ropevim/issues/91).

running `RopeRenameCurrentModule` on ubuntu 22.04, neovim 0.7.2, python 3.10, rope 1.2.0,  results in the following error:

```
Error detected while processing function RopeRenameCurrentModule[1]..provider#python3#Call:
line   18:
Error invoking 'python_execute' on channel 3 (python3-script-host):
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "~/.local/lib/python3.10/site-packages/ropemode/decorators.py", line 54, in newfunc
    return func(*args, **kwds)
  File "~/.local/lib/python3.10/site-packages/ropemode/interface.py", line 52, in do_refactor
    refactoring(self, self.env).show(initial_asking=initial_asking)
  File "~/.local/lib/python3.10/site-packages/ropemode/refactor.py", line 43, in show
    changes = runtask(self.env, calculate, name=name)
  File "~/.local/lib/python3.10/site-packages/ropemode/refactor.py", line 469, in runtask
    return RunTask(env, command, name, interrupts)()
  File "~/.local/lib/python3.10/site-packages/ropemode/refactor.py", line 489, in __call__
    result = self.task(handle)
  File "~/.local/lib/python3.10/site-packages/ropemode/refactor.py", line 41, in calculate
    return self._calculate_changes(result, handle)
  File "~/.local/lib/python3.10/site-packages/ropemode/refactor.py", line 122, in _calculate_changes
    return self.renamer.get_changes(task_handle=task_handle, **values)
  File "~/.local/lib/python3.10/site-packages/rope/refactor/rename.py", line 121, in get_changes
    new_content = rename_in_module(finder, new_name, resource=file_)
  File "~/.local/lib/python3.10/site-packages/rope/refactor/rename.py", line 235, in rename_in_module
    for occurrence in occurrences_finder.find_occurrences(resource, pymodule):
  File "~/.local/lib/python3.10/site-packages/rope/refactor/occurrences.py", line 80, in find_occurrences
    result = filter(occurrence)
  File "~/.local/lib/python3.10/site-packages/rope/refactor/occurrences.py", line 224, in __call__
    if same_pyname(self.pyname, occurrence.get_pyname()):
  File "~/.local/lib/python3.10/site-packages/rope/base/utils/__init__.py", line 12, in _wrapper
    setattr(self, name, func(self, *args, **kwds))
  File "~/.local/lib/python3.10/site-packages/rope/refactor/occurrences.py", line 147, in get_pyname
    return self.tools.name_finder.get_pyname_at(self.offset)
  File "~/.local/lib/python3.10/site-packages/rope/base/evaluate.py", line 83, in get_pyname_at
    return self.get_primary_and_pyname_at(offset)[1]
  File "~/.local/lib/python3.10/site-packages/rope/base/evaluate.py", line 87, in get_primary_and_pyname_at
    holding_scope = self.module_scope.get_inner_scope_for_offset(offset)
  File "~/.local/lib/python3.10/site-packages/rope/base/pyscopes.py", line 158, in get_inner_scope_for_offset
    return self._scope_finder.get_holding_scope_for_offset(self, offset)
  File "~/.local/lib/python3.10/site-packages/rope/base/pyscopes.py", line 315, in get_holding_scope_for_offset
    if inner_scope.in_region(offset):
  File "~/.local/lib/python3.10/site-packages/rope/base/pyscopes.py", line 116, in in_region
    region = self.get_region()
  File "~/.local/lib/python3.10/site-packages/rope/base/pyscopes.py", line 105, in get_region
    self._calculate_scope_regions_for_module()
  File "~/.local/lib/python3.10/site-packages/rope/base/pyscopes.py", line 111, in _calculate_scope_regions_for_module
    self._get_global_scope()._calculate_scope_regions()
  File "~/.local/lib/python3.10/site-packages/rope/base/utils/__init__.py", line 12, in _wrapper
    setattr(self, name, func(self, *args, **kwds))
  File "~/.local/lib/python3.10/site-packages/rope/base/pyscopes.py", line 142, in _calculate_scope_regions
    patchedast.patch_ast(self.pyobject.get_ast(), source)
  File "~/.local/lib/python3.10/site-packages/rope/refactor/patchedast.py", line 44, in patch_ast
    ast.call_for_nodes(node, walker)
  File "~/.local/lib/python3.10/site-packages/rope/base/ast.py", line 62, in call_for_nodes
    result = callback(node)
  File "~/.local/lib/python3.10/site-packages/rope/refactor/patchedast.py", line 91, in __call__
    return method(node)
  File "~/.local/lib/python3.10/site-packages/rope/refactor/patchedast.py", line 723, in _Module
    self._handle(node, list(node.body), eat_spaces=True)
  File "~/.local/lib/python3.10/site-packages/rope/refactor/patchedast.py", line 124, in _handle
    ast.call_for_nodes(child, self)
  File "~/.local/lib/python3.10/site-packages/rope/base/ast.py", line 62, in call_for_nodes
    result = callback(node)
  File "~/.local/lib/python3.10/site-packages/rope/refactor/patchedast.py", line 91, in __call__
    return method(node)
  File "~/.local/lib/python3.10/site-packages/rope/refactor/patchedast.py", line 586, in _FunctionDef
    self._handle_function_def_node(node, is_async=False)
  File "~/.local/lib/python3.10/site-packages/rope/refactor/patchedast.py", line 583, in _handle_function_def_node
    self._handle(node, children)
  File "~/.local/lib/python3.10/site-packages/rope/refactor/patchedast.py", line 124, in _handle
    ast.call_for_nodes(child, self)
  File "~/.local/lib/python3.10/site-packages/rope/base/ast.py", line 62, in call_for_nodes
    result = callback(node)
  File "~/.local/lib/python3.10/site-packages/rope/refactor/patchedast.py", line 91, in __call__
    return method(node)
  File "~/.local/lib/python3.10/site-packages/rope/refactor/patchedast.py", line 546, in _For
    self._handle_for_loop_node(node, is_async=False)
  File "~/.local/lib/python3.10/site-packages/rope/refactor/patchedast.py", line 543, in _handle_for_loop_node
    self._handle(node, children)
  File "~/.local/lib/python3.10/site-packages/rope/refactor/patchedast.py", line 124, in _handle
    ast.call_for_nodes(child, self)
  File "~/.local/lib/python3.10/site-packages/rope/base/ast.py", line 62, in call_for_nodes
    result = callback(node)
  File "~/.local/lib/python3.10/site-packages/rope/refactor/patchedast.py", line 91, in __call__
    return method(node)
  File "~/.local/lib/python3.10/site-packages/rope/refactor/patchedast.py", line 319, in _Assign
    self._handle(node, children)
  File "~/.local/lib/python3.10/site-packages/rope/refactor/patchedast.py", line 124, in _handle
    ast.call_for_nodes(child, self)
  File "~/.local/lib/python3.10/site-packages/rope/base/ast.py", line 62, in call_for_nodes
    result = callback(node)
  File "~/.local/lib/python3.10/site-packages/rope/refactor/patchedast.py", line 91, in __call__
    return method(node)
  File "~/.local/lib/python3.10/site-packages/rope/refactor/patchedast.py", line 338, in _BinOp
    children = [node.left] + self._get_op(node.op) + [node.right]
  File "~/.local/lib/python3.10/site-packages/rope/refactor/patchedast.py", line 303, in _get_op
    return self._operators[node.__class__.__name__].split(" ")
KeyError: 'MatMult'
```

### PR DESCRIPTION
**Pull request by lieryan**
# Description

Fix issue with parsing code with MatMul operator.

Fixes #496

# Checklist (delete if not relevant):

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have updated CHANGELOG.md

### CODE REVIEW

**Comment by lieryan** [comment]
Hi @eivindjahren, I wasn't able to replicate the issue, but this PR should fix this issue. Let me know if you would be able to test this branch, or when 1.3.0 is released.

**Comment by eivindjahren** [comment]
I must admit I am having a problem reproducing the error reliably also because I keep getting the following error: 
```python
Error detected while processing function RopeRenameCurrentModule[1]..provider#python3#Call:
line   18:
Error invoking 'python_execute' on channel 3 (python3-script-host):
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "~/.local/lib/python3.10/site-packages/ropemode/decorators.py", line 54, in newfunc
    return func(*args, **kwds)
  File "~/.local/lib/python3.10/site-packages/ropemode/interface.py", line 52, in do_refactor
    refactoring(self, self.env).show(initial_asking=initial_asking)
  File "~/.local/lib/python3.10/site-packages/ropemode/refactor.py", line 43, in show
    changes = runtask(self.env, calculate, name=name)
  File "~/.local/lib/python3.10/site-packages/ropemode/refactor.py", line 469, in runtask
    return RunTask(env, command, name, interrupts)()
  File "~/.local/lib/python3.10/site-packages/ropemode/refactor.py", line 489, in __call__
    result = self.task(handle)
  File "~/.local/lib/python3.10/site-packages/ropemode/refactor.py", line 41, in calculate
    return self._calculate_changes(result, handle)
  File "~/.local/lib/python3.10/site-packages/ropemode/refactor.py", line 122, in _calculate_changes
    return self.renamer.get_changes(task_handle=task_handle, **values)
  File "~/.local/lib/python3.10/site-packages/rope/refactor/rename.py", line 121, in get_changes
    new_content = rename_in_module(finder, new_name, resource=file_)
  File "~/.local/lib/python3.10/site-packages/rope/refactor/rename.py", line 235, in rename_in_module
    for occurrence in occurrences_finder.find_occurrences(resource, pymodule):
  File "~/.local/lib/python3.10/site-packages/rope/refactor/occurrences.py", line 80, in find_occurrences
    result = filter(occurrence)
  File "~/.local/lib/python3.10/site-packages/rope/refactor/occurrences.py", line 224, in __call__
    if same_pyname(self.pyname, occurrence.get_pyname()):
  File "~/.local/lib/python3.10/site-packages/rope/base/utils/__init__.py", line 12, in _wrapper
    setattr(self, name, func(self, *args, **kwds))
  File "~/.local/lib/python3.10/site-packages/rope/refactor/occurrences.py", line 147, in get_pyname
    return self.tools.name_finder.get_pyname_at(self.offset)
  File "~/.local/lib/python3.10/site-packages/rope/base/evaluate.py", line 83, in get_pyname_at
    return self.get_primary_and_pyname_at(offset)[1]
  File "~/.local/lib/python3.10/site-packages/rope/base/evaluate.py", line 87, in get_primary_and_pyname_at
    holding_scope = self.module_scope.get_inner_scope_for_offset(offset)
  File "~/.local/lib/python3.10/site-packages/rope/base/pyscopes.py", line 158, in get_inner_scope_for_offset
    return self._scope_finder.get_holding_scope_for_offset(self, offset)
  File "~/.local/lib/python3.10/site-packages/rope/base/pyscopes.py", line 315, in get_holding_scope_for_offset
    if inner_scope.in_region(offset):
  File "~/.local/lib/python3.10/site-packages/rope/base/pyscopes.py", line 117, in in_region
    return region[0] < offset < region[1]
TypeError: '<' not supported between instances of 'NoneType' and 'int'
```

**Comment by eivindjahren** [comment]
Perhaps add a regression test to `ropetest/refactor/patchedasttest.py`:
```
    def test_matmul_node(self):
        source = """
import numpy as np
a=np.array([[1,2]])
b=np.array([[1],[2]])
a @ b
"""
        ast_frag = patchedast.get_patched_ast(source, True)
        checker = _ResultChecker(self, ast_frag)
        checker.check_region("BinOp", 0, len(source) - 1)
        checker.check_children("BinOp", ["Name", " ", "@", " ", "Name"])
```
You can skip the import and use of numpy if problematic by just using lists. As I understand it, that would just result in a runtime error where it is discovered that a and b do not implement `__matmul__`.

File: rope/refactor/patchedast.py [hunk 1]
```diff
@@ -277,6 +277,7 @@ def _child_nodes(self, nodes, separator):
         "Div": "/",
         "Mod": "%",
         "Pow": "**",
+        "MatMul": "@",
```
Comment (eivindjahren) [inline rope/refactor/patchedast.py:280]: Should be `"MatMult": "@",`

Comment (eivindjahren) [reply rope/refactor/patchedast.py]: Fixed now

**Comment by eivindjahren** [comment]
The bug is fixed now!🎉 I prefer to add a regression test with every bug-fix. I find its a good way to grow test coverage. The test I pasted above should suffice. In any case, I will leave whether to included it to you.

**Comment by eivindjahren** [comment]
Nice 👍

**Pull request merged**